### PR TITLE
Fix registry name with port

### PR DESCRIPTION
**Pull request by xuezhaojun**
Signed-off-by: xuezhaojun <zxue@redhat.com>

### CODE REVIEW

File: pkg/config/agent.go [hunk 1]
```diff
@@ -15,11 +15,16 @@ func ValidateAgentImage() error {
 }
 
 func GetParsedAgentImage() (string, string, string) {
-	parts := strings.Split(AgentImageName, ":")
+	imgParts := strings.Split(AgentImageName, "/")
+	registry := strings.Join(imgParts[0:len(imgParts)-1], "/")
```
Comment (yue9944882) [inline pkg/config/agent.go:19]: defense nil panic and add test cases for the empty name

Comment (xuezhaojun) [reply pkg/config/agent.go:19]: Is it fine to change the function to `GetParsedAgentImage() (string, string, string, error)` without panic?

Comment (xuezhaojun) [reply pkg/config/agent.go:24]: Fixed.

**Comment by yue9944882** [comment]
@xuezhaojun the tests looks unhappy

**Comment by xuezhaojun** [comment]
> @xuezhaojun the tests looks unhappy

Strange blinking unit-fail occurred again, re-run for this time, having a [PR](https://github.com/open-cluster-management-io/cluster-proxy/pull/85) working on this bug currently.

**Review by yue9944882 — COMMENTED** [review]
/lgtm
/approve
